core/src/main/scala/com/vladkopanev/zio/saga/Saga.scala

```diff
@@ -51,13 +51,29 @@ final class Saga[-R, +E, +A] private (
   def zipPar[R1 <: R, E1 >: E, B](that: Saga[R1, E1, B]): Saga[R1, E1, (A, B)] =
     zipWithPar(that)((a, b) => (a, b))
 
+  /**
+   * Returns Saga that will execute this Saga in sequence with other, combining the result in a tuple.
+   * Both compensating actions would be executed in case of failure.
+   * */
+  def zip[R1 <: R, E1 >: E, B](that: Saga[R1, E1, B]): Saga[R1, E1, (A, B)] =
+    zipWith(that)((a, b) => (a, b))
+
   /**
    * Returns Saga that will execute this Saga in parallel with other, combining the result with specified function `f`.
    * Both compensating actions would be executed in case of failure.
    * */
   def zipWithPar[R1 <: R, E1 >: E, B, C](that: Saga[R1, E1, B])(f: (A, B) => C): Saga[R1, E1, C] =
     zipWithParAll(that)(f)((a, b) => a *> b)
 
+  /**
+   * Returns Saga that will execute this Saga in sequence with other, combining the result with specified function `f`.
+   * Both compensating actions would be executed in case of failure.
+   * */
+  def zipWith[R1 <: R, E1 >: E, B, C](that: Saga[R1, E1, B])(f: (A, B) => C): Saga[R1, E1, C] = (for {
+    thisResult <- this
+    thatResult <- that
+  } yield (thisResult, thatResult)).map(f.tupled)
```

Comment:
```suggestion
  } yield f(thisResult, thatResult))
```
or else it can be desugared into `.map(...).map(...)` which is kinda redundant